config_hub/pretrain/debug.yaml

```diff
@@ -1,29 +1,95 @@
+# ``model_config``. (type: Optional[str], default: null)
+model_name:
+
+# A ``litgpt.Config`` object to define the model architecture. Mutually exclusive with
+# ``model_config``. (type: Optional[Config], default: null)
+model_config:
+
+# Where to save checkpoints and logs. If running in a Lightning Studio Job, look for it in
+# /teamspace/jobs/<job-name>/share. (type: <class 'Path'>, default: out/pretrain)
+out_dir: out/pretrain
+
+# Path to a checkpoint directory to resume from in case training got interrupted. Or ``True`` to resume
+# from the latest checkpoint in ``out_dir``. (type: Union[bool, Path], default: False)
 resume: false
-devices: auto
-seed: 42
-data: OpenWebText
-out_dir: out/pretrain/debug
-tokenizer_dir: checkpoints/EleutherAI/pythia-14m
+
+# Data related arguments. If not provided, the default is ``litgpt.data.TinyLlama``. (type: Optional[LitDataModule], default: null, known subclasses: litgpt.data.LitDataModule, litgpt.data.Alpaca, litgpt.data.Alpaca2k, litgpt.data.AlpacaGPT4, litgpt.data.Dolly, litgpt.data.JSON, litgpt.data.Deita, litgpt.data.FLAN, litgpt.data.LIMA, litgpt.data.LongForm, litgpt.data.TinyLlama, litgpt.data.TinyStories, litgpt.data.OpenWebText)
+data:
+
+# raining related arguments. See ``litgpt.args.TrainArgs`` for details
 train:
+
+  # Number of optimizer steps between checkpoints (type: Optional[int], default: 1000)
   save_interval: 1000
+
+  # Number of iterations between logging calls (type: int, default: 1)
   log_interval: 1
-  global_batch_size: 125
-  micro_batch_size: 5
-  lr_warmup_steps: 100
-  epochs: null
-  max_tokens: 100000000
-  max_steps: null
-  max_seq_length: null
-  tie_embeddings: null
-  learning_rate: 6e-4
-  weight_decay: 1e-1
+
+  # Number of samples between optimizer steps across data-parallel ranks (type: int, default: 512)
+  global_batch_size: 512
+
+  # Number of samples per data-parallel rank (type: int, default: 4)
+  micro_batch_size: 4
+
+  # Number of iterations with learning rate warmup active (type: int, default: 2000)
+  lr_warmup_steps: 2000
+
+  # Number of epochs to run (type: Optional[int], default: null)
+  epochs:
+
+  # Total number of tokens to train on (type: Optional[int], default: 3000000000000)
+  max_tokens: 3000000000000
+
+  # Limits the number of optimizer steps to run. (type: Optional[int], default: null)
+  max_steps:
+
+  # Limits the length of samples. Off by default (type: Optional[int], default: null)
+  max_seq_length:
+
+  # Whether to tie the embedding weights with the language modelling head weights. (type: Optional[bool], default: False)
+  tie_embeddings: false
+
+  #   (type: float, default: 0.0004)
+  learning_rate: 0.0004
+
+  #   (type: float, default: 0.1)
+  weight_decay: 0.1
+
+  #   (type: float, default: 0.9)
   beta1: 0.9
+
+  #   (type: float, default: 0.95)
   beta2: 0.95
+
+  #   (type: Optional[float], default: 1.0)
   max_norm: 1.0
-  min_lr: 6e-5
+
+  #   (type: float, default: 4e-05)
+  min_lr: 4.0e-05
```

Comment:
why does it change these values if I regenerate the configs???

Reply:
The answer should be the same as https://github.com/Lightning-AI/litgpt/pull/1092#discussion_r1521786296